Add first pipedv1 example - Kubernetes simple app

shivansh-gohem opened the pull request:
## Description

This is the first example for PipeCD v1 demonstrating the new plugin architecture.

As requested in issue #6266, I'm creating working examples to help users understand how to use pipedv1.

## Changes

- Added `examples/pipedv1/kubernetes-simple/` directory with first pipedv1 example
- Demonstrates new `kind: Application` format (vs old `kind: KubernetesApp`)
- Shows kubernetes plugin configuration
- Includes K8S_SYNC pipeline stage
- Added comprehensive README explaining differences from piped v0
- Validated YAML syntax and Kubernetes manifests

## Files Added

- `.app.yaml` - PipeCD v1 Application configuration
- `deployment.yaml` - Kubernetes Deployment manifest
- `service.yaml` - Kubernetes Service manifest
- `README.md` - Documentation

## Next Steps

Once reviewed, I can create additional examples for:
- Canary deployments
- Helm charts
- Other plugins (terraform, cloudrun, lambda, ecs)

Addresses #6266

shivansh-gohem committed:
Add first pipedv1 example - Kubernetes simple app

- Creates first example for pipedv1 plugin architecture
- Demonstrates basic K8S_SYNC pipeline stage
- Shows new Application and plugin configuration format
- Includes documentation explaining differences from piped v0
- Validated YAML syntax and Kubernetes manifests

Addresses issue #6266

Signed-off-by: Shivansh Sahu <sahushivansh142@gmail.com>

shivansh-gohem commented on examples/pipedv1/kubernetes-simple/.app.yaml:
```diff
@@ -0,0 +1,12 @@
+    env: example
+    team: product
+  plugins:
+    - name: kubernetes
```
Thanks for the review!
At this stage, I have validated the YAML syntax and ensured the Kubernetes manifests are correct, but I have not yet tested the end-to-end deployment with a live PipeCD v1 environment.
If there are any dev resources or a recommended quickstart for spinning up pipedv1 and the control plane locally, I’d be happy to test this example end-to-end as well. Please let me know if you would like me to proceed with this, or if you have specific steps to recommend!

shivansh-gohem committed:
Fix app config format per maintainer feedback

- Rename .app.yaml to app.pipecd.yaml (required suffix)
- Fix plugins from list to map format
- Add kubernetes input configuration with namespace
- Tested YAML syntax and Kubernetes manifest validity

Note: Unable to complete full e2e testing due to network connectivity issues
downloading plugins from GitHub in this environment.

Signed-off-by: Shivansh Sahu <sahushivansh142@gmail.com>

shivansh-gohem committed:
Remove old .app.yaml file

As per maintainer feedback, removed the old .app.yaml file.
The correct file app.pipecd.yaml with proper format is already included.

Signed-off-by: Shivansh Sahu <sahushivansh142@gmail.com>

shivansh-gohem committed:
Update README per maintainer feedback

- Focus on explaining the Kubernetes simple app
- Add detailed file descriptions and purposes
- Include prerequisites and usage instructions
- Add expected deployment outcome
- Remove v0 vs v1 comparison

Addresses @eeshaanSA review feedback

Signed-off-by: Shivansh Sahu <sahushivansh142@gmail.com>

shivansh-gohem commented:
Hi @eeshaanSA! Good news - I successfully tested this example with a local PipeCD v1 setup:

**Testing Environment:**
- PipeCD v1.0.0-alpha1 control plane
- Piped agent with Kubernetes plugin v0.3.0
- kind cluster (local Kubernetes)

**Results:**
✅ Application registered successfully via PipeCD UI
✅ K8S_SYNC pipeline stage executed without errors
✅ Deployment and Service manifests applied correctly
✅ 2 nginx pods running (verified with `kubectl get pods -n pipecd`)
✅ Deployment status shows "SUCCESS" in PipeCD dashboard

The example works as expected with PipeCD v1! Previously had network issues downloading plugins, but resolved with stable connection.

Ready for your review!

eeshaanSA commented on examples/pipedv1/kubernetes-simple/README.md:
```diff
@@ -0,0 +1,90 @@
+### 2. Deploy
+
+Once registered, PipeCD will:
+1. Detect the `app.pipecd.yaml` file in this directory
```
```suggestion
1. Detect the `app.pipecd.yaml` file in this repository
```

eeshaanSA commented on examples/pipedv1/kubernetes-simple/README.md:
```diff
@@ -0,0 +1,90 @@
+
+## Next Steps
+
+Once you understand this basic example, explore:
```
This section seems unnecessary. Plus, there are no links to these resources. Plus, we do not have plugins for some apps you just mentioned. Maybe it is better to skip this section.

shivansh-gohem committed:
Address review feedback

- Remove 'Next Steps' section
- Change 'platform provider' to 'deploy target'
- Fix wording: 'directory' to 'repository'

Per @eeshaanSA feedback

Signed-off-by: Shivansh Sahu <sahushivansh142@gmail.com>

shivansh-gohem commented:
@eeshaanSA Updated! Changes made:
- ✅ Removed "Next Steps" section
- ✅ Changed "platform provider" to "deploy target" throughout
- ✅ Fixed wording: "directory" → "repository"

Here's the deployment screenshot from my testing - showing successful K8S_SYNC execution with PipeCD v1:

<img width="952" height="829" alt="Screenshot 2025-11-15 164433" src="https://github.com/user-attachments/assets/b56fcedc-7577-49d4-87fc-e135d9cd3659" />


The deployment completed successfully:
- ✅ Application registered via PipeCD UI
- ✅ K8S_SYNC pipeline stage executed without errors
- ✅ Both manifests (Deployment and Service) applied correctly
- ✅ Deployment status shows "SUCCESS"

Ready for your review!

eeshaanSA commented:
@shivansh-gohem 

https://github.com/pipe-cd/pipecd/pull/6353#issuecomment-3569971432

shivansh-gohem commented:
Is any more corrections are required ?  @eeshaanSA @khanhtc1202

khanhtc1202 commented:
Let's close this, as we have not yet decided how we can manage examples for pipecdv1 (same examples repo or separated).
Need further discussion before go with any option
